### PR TITLE
_FBTweakCollectionViewController: Observe changes in collections.

### PR DESCRIPTION
**Pull request by barakwei**
Based on #5

### CODE REVIEW

File: FBTweak/_FBTweakCollectionViewController.m [hunk 1]
```diff
@@ -17,20 +17,21 @@
...
-  NSArray *_sortedCollections;
+   NSArray<FBTweakCollection *> *_sortedCollections;
   _FBKeyboardManager *_keyboardManager;
 }
 
 - (instancetype)initWithTweakCategory:(FBTweakCategory *)category
 {
   if ((self = [super init])) {
     _tweakCategory = category;
+    self.clearsSelectionOnViewWillAppear = YES;
```
Comment (mkimi-lightricks) [inline FBTweak/_FBTweakCollectionViewController.m]: According to the documentation `clearsSelectionOnViewWillAppear`'s default value is `YES`. Why this is required?

File: FBTweak/_FBTweakCollectionViewController.m [hunk 2]
```diff
@@ -17,20 +17,21 @@
...
     self.title = _tweakCategory.name;
+    self.refreshControl = [[UIRefreshControl alloc] init];
+    [self.refreshControl addTarget:self
+                            action:@selector(_reloadData)
+                  forControlEvents:UIControlEventValueChanged];
```
Comment (mkimi-lightricks) [inline FBTweak/_FBTweakCollectionViewController.m]: Write in one line.

File: FBTweak/_FBTweakCollectionViewController.m [hunk 3]
```diff
@@ -10,12 +10,12 @@
...
 #import "FBTweak.h"
+#import "_FBKeyboardManager.h"
+#import "_FBTweakArrayViewController.h"
 #import "_FBTweakCollectionViewController.h"
-#import "_FBTweakTableViewCell.h"
 #import "_FBTweakColorViewController.h"
 #import "_FBTweakDictionaryViewController.h"
-#import "_FBTweakArrayViewController.h"
-#import "_FBKeyboardManager.h"
+#import "_FBTweakTableViewCell.h"
```
Comment (mkimi-lightricks) [inline FBTweak/_FBTweakCollectionViewController.m]: Didn't we say that we wound't do any cosmetic changes to their code?

File: FBTweak/_FBTweakCollectionViewController.m [hunk 4]
```diff
@@ -28,16 +28,33 @@ - (instancetype)initWithTweakCategory:(FBTweakCategory *)category
...
 
+- (void)dealloc
+{
+  [(id)self.tweakCategory removeObserver:self
+                              forKeyPath:NSStringFromSelector(@selector(tweakCollections))];
+}
+
+- (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(NSArray<FBTweakCollection *> *)tweakCollections change:(NSDictionary *)change context:(void *)context
+{
+  dispatch_async(dispatch_get_main_queue(), ^{
```
Comment (mkimi-lightricks) [inline FBTweak/_FBTweakCollectionViewController.m]: Enclose this block in the following construct for safety:
```objc
  if ([keyPath isEqualToString:NSStringFromSelector(@selector(tweakCollections))]) {
    // the dispatch_async block goes here.
  }
```

**Review by mkimi-lightricks — CHANGES_REQUESTED** [review]
Round.

Low priority: There are multiple places where you break the lines longer that 100 chars in code you added. Didn't we say to adhere to their coding style?

File: FBTweak/_FBTweakCollectionViewController.m (hunk 3 above)
Comment (barakwei) [reply FBTweak/_FBTweakCollectionViewController.m]: right, reverted this

File: FBTweak/_FBTweakCollectionViewController.m (hunk 1 above)
Comment (barakwei) [reply FBTweak/_FBTweakCollectionViewController.m]: removed, I'm pretty sure it was in their code somewhere, and it was needed in older versions of iOS, but I'm just guessing.

File: FBTweak/_FBTweakCollectionViewController.m (hunk 2 above)
Comment (barakwei) [reply FBTweak/_FBTweakCollectionViewController.m]: Done

File: FBTweak/_FBTweakCollectionViewController.m (hunk 4 above)
Comment (barakwei) [reply FBTweak/_FBTweakCollectionViewController.m]: Used this to do bail-fast.

**Comment by barakwei** [comment]
Round

**Comment by barakwei** [comment]
I'm not sure about breaking lines. I don't think it's that bad to break, meaning to we should not strive to fill as much in one line.
For now, I'd say that we can keep breaking line so I could stay sane but if this forces weird indentations (like in LT repos) I'll won't break in 100 chars.

Thoughts?

**Review by mkimi-lightricks — APPROVED** [review]
Squash.

**Comment by barakwei** [comment]
Squashed

**Review by mkimi-lightricks — APPROVED** [review]

**Pull request merged**